### ISSUE TITLE
heron-executor error: Unrecognized command line option: 'secret'

### ISSUE THREAD

**Issue by billonahill** [twitter/heron#2052]
I'm seeing failures in unit tests with the following `Unrecognized command line option: 'secret'` error in the heron executor. @huijunwu do you think this could be related to #2013 which added `secret` to `heron/shell/src/python/main.py`? It seems that arg is being passed to tornado, which doesn't know how to handle it.
```
[2017-07-10 01:24:32 +0000] [INFO]: Running heron-shell-2 process as ./heron-core/bin/heron-shell --port=31569 --log_file_prefix=log-files/heron-shell-2.log --secret=20170710011106_IntegrationTest_BasicTopologyOneTaskScaleUp_589d46bc-abfe-4b92-81f5-adff0968c234afc00c03-28c5-4b8e-81fc-c43a60791116
[2017-07-10 01:24:32 +0000] [INFO]: Logging pid 51351 to file heron-shell-2.pid
[2017-07-10 01:24:33 +0000] [INFO]: heron-shell-2 stdout: Usage: ./heron-core/bin/heron-shell [OPTIONS]
[2017-07-10 01:24:33 +0000] [INFO]: heron-shell-2 stdout: 
[2017-07-10 01:24:33 +0000] [INFO]: heron-shell-2 stdout: Options:
[2017-07-10 01:24:33 +0000] [INFO]: heron-shell-2 stdout: 
[2017-07-10 01:24:33 +0000] [INFO]: heron-shell-2 stdout:   --help                           show this help information
[2017-07-10 01:24:33 +0000] [INFO]: heron-shell-2 stdout:   --port                           Runs on the given port (default 9999)
[2017-07-10 01:24:33 +0000] [INFO]: heron-shell-2 stdout: 
[2017-07-10 01:24:33 +0000] [INFO]: heron-shell-2 stdout: /var/lib/mesos/slaves/5fe2f0f5-6560-468a-8a95-e52c4c5e27c9-S12107/frameworks/201104070004-0000002563-0000/executors/thermos-heron-staging-devel-20170710011106_IntegrationTest_BasicTopologyOneTaskScaleUp_589d46bc-abfe-4b92-81f5-adff0968c234-2-d61e82a8-3c7a-4210-a6f1-1a596833b004/runs/13023d41-d7a1-4fa6-a719-28db73bc5706/sandbox/.pex/install/tornado-4.0.2-py2.7-linux-x86_64.egg.2b22c54bb7558362a6cec318933489e3ff6747ac/tornado-4.0.2-py2.7-linux-x86_64.egg/tornado/log.py options:
[2017-07-10 01:24:33 +0000] [INFO]: heron-shell-2 stdout: 
[2017-07-10 01:24:33 +0000] [INFO]: heron-shell-2 stdout:   --log_file_max_size              max size of log files before rollover
[2017-07-10 01:24:33 +0000] [INFO]: heron-shell-2 stdout:                                    (default 100000000)
[2017-07-10 01:24:33 +0000] [INFO]: heron-shell-2 stdout:   --log_file_num_backups           number of log files to keep (default 10)
[2017-07-10 01:24:33 +0000] [INFO]: heron-shell-2 stdout:   --log_file_prefix=PATH           Path prefix for log files. Note that if you
[2017-07-10 01:24:33 +0000] [INFO]: heron-shell-2 stdout:                                    are running multiple tornado processes,
[2017-07-10 01:24:33 +0000] [INFO]: heron-shell-2 stdout:                                    log_file_prefix must be different for each
[2017-07-10 01:24:33 +0000] [INFO]: heron-shell-2 stdout:                                    of them (e.g. include the port number)
[2017-07-10 01:24:33 +0000] [INFO]: heron-shell-2 stdout:   --log_to_stderr                  Send log output to stderr (colorized if
[2017-07-10 01:24:33 +0000] [INFO]: heron-shell-2 stdout:                                    possible). By default use stderr if
[2017-07-10 01:24:33 +0000] [INFO]: heron-shell-2 stdout:                                    --log_file_prefix is not set and no other
[2017-07-10 01:24:33 +0000] [INFO]: heron-shell-2 stdout:                                    logging is configured.
[2017-07-10 01:24:33 +0000] [INFO]: heron-shell-2 stdout:   --logging=debug|info|warning|error|none 
[2017-07-10 01:24:33 +0000] [INFO]: heron-shell-2 stdout:                                    Set the Python log level. If 'none', tornado
[2017-07-10 01:24:33 +0000] [INFO]: heron-shell-2 stdout:                                    won't touch the logging configuration.
[2017-07-10 01:24:33 +0000] [INFO]: heron-shell-2 stdout:                                    (default info)
[2017-07-10 01:24:33 +0000] [INFO]: heron-shell-2 stdout: 
[2017-07-10 01:24:33 +0000] [INFO]: heron-shell-2 stdout: Traceback (most recent call last):
[2017-07-10 01:24:33 +0000] [INFO]: heron-shell-2 stdout:   File ".bootstrap/_pex/pex.py", line 328, in execute
[2017-07-10 01:24:33 +0000] [INFO]: heron-shell-2 stdout:   File ".bootstrap/_pex/pex.py", line 260, in _wrap_coverage
[2017-07-10 01:24:33 +0000] [INFO]: heron-shell-2 stdout:   File ".bootstrap/_pex/pex.py", line 292, in _wrap_profiling
[2017-07-10 01:24:33 +0000] [INFO]: heron-shell-2 stdout:   File ".bootstrap/_pex/pex.py", line 371, in _execute
[2017-07-10 01:24:33 +0000] [INFO]: heron-shell-2 stdout:   File ".bootstrap/_pex/pex.py", line 429, in execute_entry
[2017-07-10 01:24:33 +0000] [INFO]: heron-shell-2 stdout:   File ".bootstrap/_pex/pex.py", line 434, in execute_module
[2017-07-10 01:24:33 +0000] [INFO]: heron-shell-2 stdout:   File "/opt/ee/python/2.7/lib/python2.7/runpy.py", line 192, in run_module
[2017-07-10 01:24:33 +0000] [INFO]: heron-shell-2 stdout:     fname, loader, pkg_name)
[2017-07-10 01:24:33 +0000] [INFO]: heron-shell-2 stdout:   File "/opt/ee/python/2.7/lib/python2.7/runpy.py", line 72, in _run_code
[2017-07-10 01:24:33 +0000] [INFO]: heron-shell-2 stdout:     exec code in run_globals
[2017-07-10 01:24:33 +0000] [INFO]: heron-shell-2 stdout:   File "twitter/heron/shell/src/python/main.py", line 45, in <module>
[2017-07-10 01:24:33 +0000] [INFO]: heron-shell-2 stdout:   File "/var/lib/mesos/slaves/5fe2f0f5-6560-468a-8a95-e52c4c5e27c9-S12107/frameworks/201104070004-0000002563-0000/executors/thermos-heron-staging-devel-20170710011106_IntegrationTest_BasicTopologyOneTaskScaleUp_589d46bc-abfe-4b92-81f5-adff0968c234-2-d61e82a8-3c7a-4210-a6f1-1a596833b004/runs/13023d41-d7a1-4fa6-a719-28db73bc5706/sandbox/.pex/install/tornado-4.0.2-py2.7-linux-x86_64.egg.2b22c54bb7558362a6cec318933489e3ff6747ac/tornado-4.0.2-py2.7-linux-x86_64.egg/tornado/options.py", line 525, in parse_command_line
[2017-07-10 01:24:33 +0000] [INFO]: heron-shell-2 stdout:     return options.parse_command_line(args, final=final)
[2017-07-10 01:24:33 +0000] [INFO]: heron-shell-2 stdout:   File "/var/lib/mesos/slaves/5fe2f0f5-6560-468a-8a95-e52c4c5e27c9-S12107/frameworks/201104070004-0000002563-0000/executors/thermos-heron-staging-devel-20170710011106_IntegrationTest_BasicTopologyOneTaskScaleUp_589d46bc-abfe-4b92-81f5-adff0968c234-2-d61e82a8-3c7a-4210-a6f1-1a596833b004/runs/13023d41-d7a1-4fa6-a719-28db73bc5706/sandbox/.pex/install/tornado-4.0.2-py2.7-linux-x86_64.egg.2b22c54bb7558362a6cec318933489e3ff6747ac/tornado-4.0.2-py2.7-linux-x86_64.egg/tornado/options.py", line 254, in parse_command_line
[2017-07-10 01:24:33 +0000] [INFO]: heron-shell-2 stdout:     raise Error('Unrecognized command line option: %r' % name)
[2017-07-10 01:24:33 +0000] [INFO]: heron-shell-2 stdout: tornado.options.Error: Unrecognized command line option: 'secret'
```

**Comment by huijunw** [comment]
It is twitter internal repo sync issue rather than a bug in github repo.

**Comment by objmagic** [comment]
I have a PR ready and will test it later.

**Comment by objmagic** [comment]
should be closed via #2054 and an internal patch.
